controller/share_manager_controller.go

```diff
@@ -257,6 +257,9 @@ func (c *ShareManagerController) handleErr(err error, key interface{}) {
 func (c *ShareManagerController) syncShareManager(key string) (err error) {
 	defer func() {
 		err = errors.Wrapf(err, "failed to sync %v", key)
+		if err != nil {
+			c.logger.WithError(err).Errorf("Failed to sync share manager %v", key)
+		}
```

Comment:
Doesn't the error be logged in the below code already?

[/controller/utils.go#L72](https://github.com/longhorn/longhorn-manager/blob/b5e478db9114a9b47f6f4b0bff916a3e4568b0f3/controller/utils.go#L72)
```go
		logger.WithError(err).Error(mesg)
```

Reply:
Ah... you're right.

BTW, 
```
	if apierrors.IsConflict(err) {
		logger.WithError(err).Debug(mesg)
```
Does using `Debug` for conflict make sense?

Reply:
After discussing with @derekbit , we decided to change the log level to warn, so it would be still helpful for us to know if there is any failed resource update to cause the resource status not updated successfully.